packages/world/src/ObjectType.sol

```diff
@@ -0,0 +1,925 @@
+
+  function isOreBlock(ObjectType self) internal pure returns (bool) {
+    return category(self) == Category.OreBlock;
+  }
+
+  function isSeed(ObjectType self) internal pure returns (bool) {
+    return category(self) == Category.Seed;
+  }
+
+  function isSapling(ObjectType self) internal pure returns (bool) {
+    return category(self) == Category.Sapling;
+  }
+
+  function isStation(ObjectType self) internal pure returns (bool) {
+    return category(self) == Category.Station;
+  }
+
+  function isSmartEntityBlock(ObjectType self) internal pure returns (bool) {
+    return category(self) == Category.SmartEntityBlock;
+  }
+
+  function isMiscBlock(ObjectType self) internal pure returns (bool) {
+    return category(self) == Category.MiscBlock;
+  }
+
+  function isPick(ObjectType self) internal pure returns (bool) {
+    return category(self) == Category.Pick;
+  }
+
+  function isAxe(ObjectType self) internal pure returns (bool) {
+    return category(self) == Category.Axe;
+  }
+
+  function isHoe(ObjectType self) internal pure returns (bool) {
+    return category(self) == Category.Hoe;
+  }
+
+  function isWhacker(ObjectType self) internal pure returns (bool) {
+    return category(self) == Category.Whacker;
+  }
+
+  function isOreBar(ObjectType self) internal pure returns (bool) {
+    return category(self) == Category.OreBar;
+  }
+
+  function isBucket(ObjectType self) internal pure returns (bool) {
+    return category(self) == Category.Bucket;
+  }
+
+  function isFood(ObjectType self) internal pure returns (bool) {
+    return category(self) == Category.Food;
+  }
+
+  function isFuel(ObjectType self) internal pure returns (bool) {
+    return category(self) == Category.Fuel;
+  }
+
+  function isMovable(ObjectType self) internal pure returns (bool) {
+    return category(self) == Category.Movable;
+  }
+
+  function isSmartEntityNonBlock(ObjectType self) internal pure returns (bool) {
+    return category(self) == Category.SmartEntityNonBlock;
+  }
+
+  // Category getters
+  function getNonSolidTypes() internal pure returns (ObjectType[4] memory) {
+    return [ObjectTypes.Null, ObjectTypes.Air, ObjectTypes.Water, ObjectTypes.Lava];
+  }
+
+  function getStoneTypes() internal pure returns (ObjectType[16] memory) {
+    return [
+      ObjectTypes.Stone,
+      ObjectTypes.Bedrock,
+      ObjectTypes.Deepslate,
+      ObjectTypes.Granite,
+      ObjectTypes.Tuff,
+      ObjectTypes.Calcite,
+      ObjectTypes.Basalt,
+      ObjectTypes.SmoothBasalt,
+      ObjectTypes.Andesite,
+      ObjectTypes.Diorite,
+      ObjectTypes.Cobblestone,
+      ObjectTypes.MossyCobblestone,
+      ObjectTypes.Obsidian,
+      ObjectTypes.Dripstone,
+      ObjectTypes.Blackstone,
+      ObjectTypes.CobbledDeepslate
+    ];
+  }
+
+  function getGemstoneTypes() internal pure returns (ObjectType[2] memory) {
+    return [ObjectTypes.Amethyst, ObjectTypes.Glowstone];
+  }
+
+  function getSoilTypes() internal pure returns (ObjectType[9] memory) {
```

Comment:
This lib is quite big, does forge only include functions that a system uses or all of this bytecode?

Reply:
yes, internal functions are inlined so it can be as big as we need